bin/src/data/csv.py

```diff
@@ -105,23 +124,41 @@ def add_split(self, config: dict,  force=False) -> None:
         split_column[test] = 2
         self.data = self.data.with_columns(pl.Series('split:split:int', split_column))
 
-    def add_noise(self, configs: list) -> None:
+    def transform(self, transformations: list) -> None:
         """
-        Adds noise to the data.
-        Noise is added for each column with the specified configurations.
+        Transforms the data using the specified configuration.
         """
-        # for each column configuration
-        for dictionary in configs:
+        for dictionary in transformations:
             key = dictionary['column_name']
             data_type = key.split(':')[2]
-            noise_generator = dictionary['name']
-
-            # add noise to the column using the desired noise generator and params
-            new_column = self.experiment.get_function_noise_all(data_type, noise_generator)(list(self.data[key]), **dictionary['params'])
-
-            # change the column with the new values
-            self.data = self.data.with_columns(pl.Series(key, new_column))
-
+            data_transformer = dictionary['name']
+            transfomer = self.experiment.get_data_transformer(data_type, data_transformer)
+            
+            # If the transformer is only for training data, we need to separate the data
+            # and transform only the training data
+            if transfomer.training_data_only:
+                data_to_transform = self.data.filter(pl.col('split:split:int') == 0)
```

Comment:
is it possible to rely only of the presence of the word "split" instead of the whole column anme. more robust.

Reply:
done